applications/GeoMechanicsApplication/custom_processes/find_neighbour_elements_of_conditions_process.cpp

```diff
@@ -8,406 +8,193 @@
+                                                                              const std::vector<std::size_t>& rSecond)
 {
-    KRATOS_TRY
-
-    hashmap::iterator itFace = FacesMap.find(FaceIds);
-    if (itFace != FacesMap.end()) return itFace;
-
-    if (FaceIds.size() == 8) {
-        // first try
-        std::swap(FaceIds[0], FaceIds[1]);
-        std::swap(FaceIds[1], FaceIds[2]);
-        std::swap(FaceIds[2], FaceIds[3]);
-        std::swap(FaceIds[4], FaceIds[5]);
-        std::swap(FaceIds[5], FaceIds[6]);
-        std::swap(FaceIds[6], FaceIds[7]);
+    const auto amount_of_needed_rotations = std::ranges::find(First, rSecond[0]) - First.begin();
 
-        itFace = FacesMap.find(FaceIds);
-        if (itFace != FacesMap.end()) return itFace;
+    std::rotate(First.begin(), First.begin() + amount_of_needed_rotations, First.begin() + First.size() / 2);
 
-        // Second try
-        std::swap(FaceIds[0], FaceIds[1]);
-        std::swap(FaceIds[1], FaceIds[2]);
-        std::swap(FaceIds[2], FaceIds[3]);
-        std::swap(FaceIds[4], FaceIds[5]);
-        std::swap(FaceIds[5], FaceIds[6]);
-        std::swap(FaceIds[6], FaceIds[7]);
+    std::rotate(First.begin() + First.size() / 2,
+                First.begin() + First.size() / 2 + amount_of_needed_rotations, First.end());
 
-        itFace = FacesMap.find(FaceIds);
-        if (itFace != FacesMap.end()) return itFace;
+    return First == rSecond;
+}
 
-        // Third try
-        std::swap(FaceIds[0], FaceIds[1]);
-        std::swap(FaceIds[1], FaceIds[2]);
-        std::swap(FaceIds[2], FaceIds[3]);
-        std::swap(FaceIds[4], FaceIds[5]);
-        std::swap(FaceIds[5], FaceIds[6]);
-        std::swap(FaceIds[6], FaceIds[7]);
+bool FindNeighbourElementsOfConditionsProcess::AllConditionsHaveAtLeastOneNeighbour() const
+{
+    return std::ranges::all_of(mrModelPart.Conditions(), [](auto& rCondition) {
+        return rCondition.GetValue(NEIGHBOUR_ELEMENTS).size() >= 1;
+    });
+}
 
-        itFace = FacesMap.find(FaceIds);
-        if (itFace != FacesMap.end()) return itFace;
+void FindNeighbourElementsOfConditionsProcess::ReportConditionsWithoutNeighbours() const
+{
+    std::vector<Condition> conditions_without_neighbours;
+    std::ranges::copy_if(
+        mrModelPart.Conditions(), std::back_inserter(conditions_without_neighbours),
+        [](const auto& rCondition) { return rCondition.GetValue(NEIGHBOUR_ELEMENTS).size() == 0; });
+    for (const auto& r_condition : conditions_without_neighbours) {
+        KRATOS_INFO("Condition without any corresponding element, ID ") << r_condition.Id() << std::endl;
     }
+}
 
-    return FacesMap.end();
-
-    KRATOS_CATCH("")
+std::vector<std::size_t> FindNeighbourElementsOfConditionsProcess::GetNodeIdsFromGeometry(const Geometry<Node>& rGeometry)
+{
+    std::vector<std::size_t> result(rGeometry.size());
+    std::ranges::transform(rGeometry, result.begin(), [](const auto& rNode) { return rNode.Id(); });
```

Comment:
To avoid creating a result vector with uninitialized elements, I generally prefer to create an empty vector, and then set its capacity:
```suggestion
    auto result = std::vector<std::size_t>{};
    result.reserve(rGeometry.size());
    std::ranges::transform(rGeometry, std::back_inserter(result), [](const auto& rNode) { return rNode.Id(); });
```

Reply:
Yes good point, also more consistent with our other code